cli/internal/scm/scm.go

```diff
@@ -51,3 +53,23 @@ func FromInRepo(repoRoot turbopath.AbsoluteSystemPath) (SCM, error) {
 	}
 	return newFallback(dotGitDir.Dir())
 }
+
+// GetCurrentBranch returns the current branch
+func GetCurrentBranch() string {
+	command := []string{"rev-parse", "--abbrev-ref", "HEAD"}
```

Comment:
This can pull a lot of surprising information with it.

Detached HEAD:
```
cd turbo
git checkout 2b0654e0e19da62547f968fb95bb2d0fe982ca1c
git rev-parse --abbrev-ref HEAD
# HEAD
```

Here is a branch and a tag with the same name:
```
~/repos/triage/4734 (master)$ git log
commit 91ae59e50f922accfca0b052db6446ab54e3a051 (HEAD -> master, tag: master, tag: foo)
Author: Nathan Hammond <nathan.hammond@vercel.com>
Date:   Fri Apr 28 15:18:20 2023 +0800

    initial commit
~/repos/triage/4734 (master)$ git rev-parse --abbrev-ref HEAD
# heads/master
```

You may want to look into `--symbolic-full-name` to get consistent behavior.

Bonus points:
- Does this work with worktrees?
- What about submodules?

Reply:
Switched to `git branch --show-current`, any concerns with that?

Reply:
Nope: https://github.com/git/git/blob/master/builtin/branch.c#L500-L513